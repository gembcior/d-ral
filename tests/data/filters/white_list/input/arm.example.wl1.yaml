peripherals:
  - name: TIMER0
    registers:
      - name: CR
      - name: SR
      - name: INT
