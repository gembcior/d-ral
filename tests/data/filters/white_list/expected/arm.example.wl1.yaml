description: 'ARM 32-bit Cortex-M3 Microcontroller based device, CPU clock up to 80MHz, etc. '
name: ARM_Example
peripherals:
  - address: 1073807360
    description: 32 Timer / Counter, counting up or down from different sources
    name: TIMER0
    registers:
    - access: read-write
      description: Control Register
      fields:
      - description: Enable
        mask: 1
        name: EN
        position: 0
        width: 1
      - description: Reset Timer
        mask: 1
        name: RST
        position: 1
        width: 1
      - description: Counting direction
        mask: 3
        name: CNT
        position: 2
        width: 2
      - description: Operation Mode
        mask: 7
        name: MODE
        position: 4
        width: 3
      - description: Use Prescaler
        mask: 1
        name: PSC
        position: 7
        width: 1
      - description: Timer / Counter Source Divider
        mask: 15
        name: CNTSRC
        position: 8
        width: 4
      - description: Timer / Counter Capture Source
        mask: 15
        name: CAPSRC
        position: 12
        width: 4
      - description: Capture Edge, select which Edge should result in a counter increment
          or decrement
        mask: 3
        name: CAPEDGE
        position: 16
        width: 2
      - description: Triggers an other Peripheral
        mask: 3
        name: TRGEXT
        position: 20
        width: 2
      - description: Select RELOAD Register n to reload Timer on condition
        mask: 3
        name: RELOAD
        position: 24
        width: 2
      - description: Selects, if Reload Register number is incremented, decremented
          or not modified
        mask: 3
        name: IDR
        position: 26
        width: 2
      - description: Starts and Stops the Timer / Counter
        mask: 1
        name: S
        position: 31
        width: 1
      name: CR
      offset: 0
      reset_value: 0
      size: 32

    - access: read-write
      description: Status Register
      fields:
      - description: Shows if Timer is running or not
        mask: 1
        name: RUN
        position: 0
        width: 1
      - description: Shows if the MATCH was hit
        mask: 1
        name: MATCH
        position: 8
        width: 1
      - description: Shows if an underflow occured. This flag is sticky
        mask: 1
        name: UN
        position: 9
        width: 1
      - description: Shows if an overflow occured. This flag is sticky
        mask: 1
        name: OV
        position: 10
        width: 1
      - description: Shows if Timer is in RESET state
        mask: 1
        name: RST
        position: 12
        width: 1
      - description: Shows the currently active RELOAD Register
        mask: 3
        name: RELOAD
        position: 14
        width: 2
      name: SR
      offset: 4
      reset_value: 0
      size: 16

    - access: read-write
      description: Interrupt Register
      fields:
      - description: Interrupt Enable
        mask: 1
        name: EN
        position: 0
        width: 1
      - description: Interrupt Mode, selects on which condition the Timer should generate
          an Interrupt
        mask: 7
        name: MODE
        position: 4
        width: 3
      name: INT
      offset: 16
      reset_value: 0
      size: 16
